### DISCUSSION TITLE
Best way to run custom apps and stop them

### DISCUSSION THREAD

**Discussion by Cyb3r-Monk** [cmu-sei/GHOSTS#188] [Q&A] [answered]
I want to start an app, let's say Spotify, let it run for a while and then close it. What would be the best way to do it? I can use the command handler to start the app, but how about stopping it? Should I just find the process name and use taskkill or something? For every single application, I would have to have two handlers in this case.

**Comment by sei-dupdyke** [comment, marked as answer]
It depends—the agent would likely not want to stop if the song is sufficiently good 😁

There are likely infinite ways to do this, I'm not sure I know the best/easiest way. Is this an argument to build out AutoIT? 

You could write a bat script to start it, sleep, and then kill it after x minutes. Then the ghosts handler just calls cmd execute on that batch (or powershell) file.

**Reply by Cyb3r-Monk** [reply]
I didn't think about the AutoIT. Batch/PS scripts sounds like a plan!
